kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/notification/NotificationManager.kt

```diff
@@ -272,26 +277,31 @@ class NotificationManager internal constructor(private val context: Context, pri
         }
     }
 
-    internal fun onPlay() {
+    internal fun onPlay() = scope.launch {
         mediaSession.isActive = true
         reload()
     }
 
-    internal fun onPause() {
+    internal fun onPause() = scope.launch {
         reload()
     }
 
-    internal fun destroy() {
-        descriptionAdapter?.release()
+    fun onUnbind() = scope.launch {
+        reload()
+    }
+
+    internal fun destroy() = scope.launch {
+        mediaSession.isActive = false
+        descriptionAdapter.release()
         internalManager?.setPlayer(null)
     }
```

Comment:
Notification processes must be run from the main thread. Previously it was causing issues where some processes would not finish in time. 
I added a `scope.launch` to every notification function for extra thread safety.